### PR TITLE
Release Automation: Push images on cloud

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: build/release/cloudbuild.yaml [hunk 1]
```diff
@@ -26,21 +53,96 @@ steps:
...
+  # Build all the images
+  #
+  - name: "make-docker"
+    id: build-images
+    dir: "build"
+    env:
+      - "VERSION=${_VERSION}"
+      - "REGISTRY=${_REGISTRY}/release"
+      - "FULL_BUILD=1"
+    args: ["-j", "1", "build"]
```
Comment (markmandel) [inline build/release/cloudbuild.yaml:89]: Could we do this in parrallel? (i.e. `-j=4` (or whatever number is the best)? Maybe make things a bit faster?)

File: build/release/cloudbuild.yaml [hunk 2]
```diff
@@ -26,21 +53,96 @@ steps:
...
+  #
+  - name: "make-docker"
+    id: push-images
+    waitFor:
+      - build-images
+    dir: "build"
+    env:
+      - "VERSION=${_VERSION}"
+      - "REGISTRY=${_REGISTRY}/release"
+    args: ["-j", "1", "push"]
```
Comment (markmandel) [inline build/release/cloudbuild.yaml:102]: Could we do this in parrallel? (i.e. `-j=4` (or whatever number is the best)? Maybe make things a bit faster?)

**Review by markmandel — COMMENTED** [review]
Looks really good!

Let's also take these step out of `make do-release` and this is probably good to go.

Unless we need to update the release checklist as well? You probably know this best at this point 👍🏻 

https://github.com/googleforgames/agones/blob/02e57babccab6489b18d956286152174fe9a1eaa/build/includes/release.mk#L74-L96

File: build/release/cloudbuild.yaml (hunk 1 above)
Comment (Kalaiselvi84) [reply build/release/cloudbuild.yaml:89]: Tried increasing jobs from 1 to 12. Build is succeeded with 1 job only. Received error if I increase the jobs -> docker: Error response from daemon: pull access denied for agones-build, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.

**Comment by Kalaiselvi84** [comment]
> $(MAKE) push-chart VERSION=$(RELEASE_VERSION)

This step is not required in cloud?

File: build/release/cloudbuild.yaml (hunk 1 above)
Comment (markmandel) [reply build/release/cloudbuild.yaml:89]: Fair enough. At some point we may want to work out why it's doing that.

**Review by markmandel — COMMENTED** [review]
Sorry - sent you a message offline - just realised I've sent you down the wrong path - my apologies.

**Comment by Kalaiselvi84** [comment]
> sent you a message offline

I am working on it..

File: build/includes/release.mk [hunk 3]
```diff
@@ -65,36 +65,14 @@ release-deploy-site:
...
-	-rm -rf $(agones_path)/release
-	mkdir $(agones_path)/release
-
-	$(MAKE) -j 4 build VERSION=$(RELEASE_VERSION) REGISTRY=$(release_registry) FULL_BUILD=1
-	cp $(agones_path)/cmd/sdk-server/bin/agonessdk-server-$(RELEASE_VERSION).zip $(agones_path)/release
-	cp $(agones_path)/sdks/cpp/.archives/agonessdk-$(RELEASE_VERSION)-linux-arch_64.tar.gz $(agones_path)/release
-	cd $(agones_path) &&  zip -r ./release/agones-install-$(RELEASE_VERSION).zip ./README.md ./install ./LICENSE
-
-	$(MAKE) gcloud-auth-docker
-	$(MAKE) -j 4 push REGISTRY=$(release_registry) VERSION=$(RELEASE_VERSION)
```
Comment (markmandel) [inline build/includes/release.mk:91]: Let's keep the target, but remove the building of the images and pushing them.

We still need to push the helm chart (which is being worked on), and also work out where to sort out the git branch work as well.

File: build/release/post_cloudbuild.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
+# Copyright 2021 Google LLC
```
Comment (markmandel) [inline build/release/post_cloudbuild.yaml:1]: New files, so 
```suggestion
# Copyright 2023 Google LLC
```

**Review by markmandel — COMMENTED** [review]
Couple of small things, but this looks good 👍🏻

File: build/includes/release.mk (hunk 3 above)
Comment (Kalaiselvi84) [reply build/includes/release.mk:91]: Included the do-release target with push-chart and git branch. Will work on the branch part along with push-chart target in b/278275965

File: build/release/post_cloudbuild.yaml (hunk 4 above)
Comment (Kalaiselvi84) [reply build/release/post_cloudbuild.yaml]: fixed it 👍🏻

**Review by markmandel — APPROVED** [review]

**Pull request merged**